src/App.tsx

```diff
@@ -1,14 +1,66 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
+import { Options } from './types/Options';
+import { getTodos } from './api';
 import { Loader } from './components/Loader';
 
+const getFilteredTodos = (todos: Todo[], filter: Filter):Todo[] => {
+  let filterTodos = [...todos];
+
+  filterTodos = filterTodos.filter(todo => {
+    switch (filter.option) {
+      case Options.Active:
+        return !todo.completed;
+      case Options.Completed:
+        return todo.completed;
+      case Options.All:
+      default:
+        return todo;
+    }
+  });
+
+  if (filter.query) {
+    const lowercaseQuery = filter.query.toLowerCase();
+
+    filterTodos = filterTodos
+      .filter(({ title }) => title.toLowerCase().includes(lowercaseQuery));
+  }
+
+  return filterTodos;
+};
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [task, setTask] = useState<Todo | null>(null);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [filter, setFilter] = useState<Filter>({ option: Options.All, query: '' });
+
+  const filteredTodos = getFilteredTodos(todos, filter);
+
+  useEffect(() => {
+    getTodos().then(setTodos)
+      .finally(() => setLoading(false)).catch((error) => {
+      // eslint-disable-next-line no-console
+        console.error('Something bad happened!', error);
+      });
```

Comment:
keep your code consistent
```suggestion
    getTodos()
      .then(setTodos)
      .catch(error => {
        // eslint-disable-next-line no-console
        console.error(error);
      })
      .finally(() => setLoading(false))
```